bitnami/harbor/templates/core/core-secret.yaml

```
@@ -3,6 +3,7 @@ Copyright Broadcom, Inc. All Rights Reserved.
 SPDX-License-Identifier: APACHE-2.0
 */}}
 
+{{- if or (not .Values.core.existingSecret) (not .Values.core.secretName) }}
```

This two variables are not use fot the same purpose (a bit confusing I think). I suggest this change:
```suggestion
{{- if (not .Values.core.existingSecret) }}
```

---

You are correct that they are used for different things, however, if you check the data section, you'll notice that if both secrets are provided, the secret in this file will be empty. While this is not technically "wrong", it does mean that it A) still creates a secret (when the goal of the PR is to enable it to output zero secrets) and B) prevents the user from creating a secret out-of-band with the same name (as the chart would overwrite it). By checking if at least one of those secrets is not already provided, we can avoid outputting an unnecessary empty secret.

Let me know if I've misunderstood your comment.